tests/bugzilla_test.py

```diff
@@ -0,0 +1,162 @@
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+    rsp = client.post('/endpoint/bugzilla/' + TOKEN + '/' + PRODUCT, data=payload, content_type="application/json")
+    assert rsp.status == '405 METHOD NOT ALLOWED'
+    rsp = client.post('/endpoint/bugzilla/' + TOKEN + '/' + PRODUCT + '/' + COMPONENT, data=payload, content_type="application/json")
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+    rsp = client.post('/endpoint/bugzilla/' + TOKEN + '/' + PRODUCT + '/' + COMPONENT + '/' + VERSION, data=payload, content_type="application/json")
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+    rsp = client.post('/endpoint/bugzilla/' + PRODUCT, data=payload, content_type="application/json")
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+    rsp = client.post('/endpoint/bugzilla/' + PRODUCT + '/' + COMPONENT, data=payload, content_type="application/json")
+    assert rsp.status == '405 METHOD NOT ALLOWED'
+    rsp = client.post('/endpoint/bugzilla/' + PRODUCT + '/' + COMPONENT + '/' + VERSION, data=payload, content_type="application/json")
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+
+
+def test_bugzilla_wrongpass():
+    loginsightwebhookdemo.bugzilla.BUGZILLAURL = 'https://landfill.bugzilla.org/bugzilla-5.0-branch/'
+    loginsightwebhookdemo.bugzilla.BUGZILLAUSER = 'loginsight-marketplace@vmware.com'
+    loginsightwebhookdemo.bugzilla.BUGZILLAPASS = 'wrongpass'
+    loginsightwebhookdemo.bugzilla.BUGZILLAPRODUCT = 'MyOwnBadSelf'
+    loginsightwebhookdemo.bugzilla.BUGZILLACOMPONENT = 'comp2'
+    loginsightwebhookdemo.bugzilla.BUGZILLAVERSION = 'unspecified'
+    loginsightwebhookdemo.bugzilla.VERIFY = True
+    payload = generate_alertname()
+
+    rsp = client.post('/endpoint/bugzilla/' + TOKEN, data=payload, content_type="application/json")
+    assert rsp.status == '200 OK'
+
+    payload = generate_alertname()
+
+    rsp = client.post('/endpoint/bugzilla/' + TOKEN + '/' + PRODUCT + '/' + COMPONENT + '/' + VERSION, data=payload, content_type="application/json")
+    assert rsp.status == '200 OK'
+
+    payload = generate_alertname()
+
+    rsp = client.post('/endpoint/bugzilla/' + PRODUCT + '/' + COMPONENT + '/' + VERSION, data=payload, content_type="application/json")
+    assert rsp.status == '401 UNAUTHORIZED'
+
+
+def test_bugzilla_nopass():
+    loginsightwebhookdemo.bugzilla.BUGZILLAPASS = ''
+    payload = generate_alertname()
+
+    rsp = client.post('/endpoint/bugzilla/' + TOKEN, data=payload, content_type="application/json")
+    assert rsp.status == '200 OK'
+    rsp = client.post('/endpoint/bugzilla', data=payload, content_type="application/json")
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+    rsp = client.post('/endpoint/bugzilla/' + PRODUCT + '/' + COMPONENT + '/' + VERSION, data=payload, content_type="application/json")
+    assert rsp.status == '500 INTERNAL SERVER ERROR'
+
+
+def test_bugzilla_wrongtoken():
+    loginsightwebhookdemo.bugzilla.BUGZILLAPASS = 'L0g1ns1ght!'
```

Comment:
Same as token -- goal is to test code + provide information tester would need in order to validate.